tests/fast-integration/skr-svcat-migration-test/test-helpers.js

```diff
@@ -63,6 +69,85 @@ async function installBTPServiceOperatorMigrationHelmChart() {
     }
 }
 
+async function getFunctionPod(functionName) {
+    let labelSelector = `serverless.kyma-project.io/function-name=${functionName},serverless.kyma-project.io/resource=deployment`;
+    let res = {};
+    for (let i = 0; i < 30; i++) {
+        res = await listPods(labelSelector);
+        if (res.body.items.length == 1) {
+            let pod = res.body.items[0];
+            if (pod.status.phase == "Running") {
+                return pod
+            }
+        }
+        sleep(10000);
+    }
+    if (res.body.items.length != 1) {
+        let podNames = res.body.items.map(p=>p.metadata.name);
+        let phases = res.body.items.map(p=>p.status.phase);
+        throw new Error(`Failed to find function ${functionName} pod in 5 minutes. Expected 1 ${labelSelector} pod with phase "Runninng" but found ${res.body.items.length}, ${podNames}, ${phases}`);
+    }
+}
+
+async function checkPodPresetEnvInjected() {
+    let cmd = `for v in {vars}; do x="$(eval echo \\$$v)"; if [[ -z "$x" ]]; then echo missing $v env variable; exit 1; else echo found $v env variable; fi; done`;
+    for (let f of functions) {
+        let pod = await getFunctionPod(f.name);
+        let envCmd = cmd.replace("{vars}",f.checkEnvVars);
+        await kubectlExecInPod(pod.metadata.name, "function", ["sh", "-c", envCmd]);
```

Comment:
Is it worth catching potential errors here?
What happens if Pod is not running anymore (for whatever reason) or network hiccup kicks?

Reply:
the `execa` throws an error inside javascript if exit code is non-zero. And `kubectlExecInPod` uses `kubectl` inside `execa`. So if `kubectl exec -ti [pod] -- cmd` fails (which in case of network or `Pod` disappearing it will`, it will throw an error and this just propagates it up the call stack to the point it's visible in mocha logs and the particular `It()` step marked as failed.

Similarly, if the env variable is not injected for some reason by `PodPresets`, the command will `exit 1` and print which env vars are present and which are missing.